geocoded_object_extractor/geocoded_object_extractor.py

```diff
@@ -72,7 +72,8 @@ def _generate_cutouts_for_image(self, rgb):
 
             # some polygons may have very small intersections
             try:
-                cutout = rgb.rio.clip([bbox], drop=True)
+                cutout_box = rgb.rio.clip_box(*bbox.bounds) # First clip with bounding box
```

Comment:
I guess this is to improve performance in case of a COG/blocked file structure, is this correct? Maybe we could just add a comment explaining why we do that - just something like it is more efficient to first do the cropping based on the bounds and then use the actual geometry?

Reply:
Nice idea! I added the comment you suggested